eng/pipelines/build.yml

```diff
@@ -128,62 +128,88 @@ jobs:
       - _HelixSource: pr/dotnet/arcade/$(Build.SourceBranch)
 
     # This is only required for cross builds.
-    - ${{ if and(eq(parameters.crossbuild, false), eq(parameters.crossrootfsDir, '')) }}:
-      - _Cross: ''
-    - ${{ if or(eq(parameters.crossbuild, true), ne(parameters.crossrootfsDir, '')) }}:
+    - ${{ if eq(parameters.crossBuild, true) }}:
       - _Cross: -cross
 
+    # If there is a native build container, build managed in the host vm/container and native in the nativeBuildContainer
+    - ${{ if ne(parameters.nativeBuildContainer, '') }}:
+      - _ExtraBuildParams: -skipnative
+      - _BuildDisplayName: 'Build Managed'
+
+    # Only add the cross build option if a combined build/test managed/native build (i.e. MacOS arm64)
+    - ${{ if eq(parameters.nativeBuildContainer, '') }}:
+      - _ExtraBuildParams: $(_Cross)
+
     steps:
     - ${{ if eq(parameters.osGroup, 'Linux') }}:
       - ${{ if eq(parameters.testOnly, 'true') }}:
         - task: DownloadBuildArtifacts@0
-          displayName: 'Download release builds'
+          displayName: 'Download Build Artifacts'
           inputs:
             downloadPath: '$(Build.ArtifactStagingDirectory)/__download__'
-            downloadType: specific 
+            downloadType: specific
             itemPattern: |
-              Build_$(_BuildConfig)/bin/Linux.$(_BuildArch).$(_BuildConfig)/**
+              Build_$(_BuildConfig)/bin/Linux${{ parameters.osSuffix }}.$(_BuildArch).$(_BuildConfig)/**
             checkDownloadedFiles: true
         - task: CopyFiles@2
           displayName: 'Binplace Product'
           inputs:
-            sourceFolder: $(Build.ArtifactStagingDirectory)/__download__/Build_$(_BuildConfig)/bin/Linux.$(_BuildArch).$(_BuildConfig)
+            sourceFolder: $(Build.ArtifactStagingDirectory)/__download__/Build_$(_BuildConfig)/bin/Linux${{ parameters.osSuffix }}.$(_BuildArch).$(_BuildConfig)
             targetFolder: '$(Build.SourcesDirectory)/artifacts/bin/Linux.$(_BuildArch).$(_BuildConfig)'
 
     - ${{ if eq(parameters.isCodeQLRun, 'true') }}:
       - task: CodeQL3000Init@0
         displayName: CodeQL Initialize
 
-    - script: $(_dockerEnv) $(_buildScript)
-        -configuration $(_BuildConfig) 
+    - script: $(_buildScript)
+        -ci
+        -configuration $(_BuildConfig)
         -architecture $(_BuildArch)
-        $(_Cross)
+        $(_ExtraBuildParams)
         $(_TestArgs)
         /p:OfficialBuildId=$(BUILD.BUILDNUMBER)
         $(_InternalInstallArgs)
-      displayName: Build / Test
+      displayName: $(_BuildDisplayName)
```

Comment:
This is the only usage of $(_BuildDisplayName). Can we inline the usage of it with expansion directly here? Variable usage should be limited to where they are needed - they bind late and make diagnosing pipeline issues unnecessarily hard.

Reply:
It is either "Build / Test" for the combined build/test legs or "Build Managed" for the managed only legs. I wanted to see this difference in the step names.

Reply:
That's fine - but a variable is not necessary